tests/swap_test.go

```diff
@@ -0,0 +1,293 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"fmt"
+	"time"
+
+	expect "github.com/google/goexpect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/utils/pointer"
+
+	v1 "k8s.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/tests/console"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/tests"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+const (
+	// Define relevant k8s versions
+	k8sSwapVer = "1.22"
+	//20% of the default size which is 2G in kubevirt-ci
+	maxSwapSizeToUsekb = 415948
+	swapPartToUse      = 0.2
+
+	gigbytesInkb = 1000000
```

Comment:
Lots of `KB` vs `KiB` confusion (at least for me!) in this file.
Please make sure that everything is coherent.
I believe everything should be in `*iB`, in which case the multiplier used throughout the tests should be `1024`, not `1000`.
This could add up, as there's a ~70MiB difference between 1GB and 1GiB!

Reply:
i see you point 
i modified the numbers to refer only to Kib
hope it's less confusing now